### PR TITLE
[ga] Clarify test for dhá/beirt (etc.)

### PR DESCRIPTION
**Pull request by jimregan**

### CODE REVIEW

File: languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java [hunk 1]
```diff
@@ -65,9 +65,9 @@ public String getDescription() {
     String replacement = null;
     String msg = null;
     for (int i = 1; i < tokens.length; i++) {  // ignoring token 0, i.e., SENT_START
-      if (isNumber(tokens[i]) && (i < tokens.length - 2 && isPerson(tokens[i + 1]))) {
+      if (isNumber(tokens[i]) && (i < tokens.length - 1 && isPerson(tokens[i + 1]))) {
```
Comment (TiagoSantos81) [inline languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java:68]: @jimregan Whitout this you still get IndexOutOfBounds due to `j=i+2`

File: languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java [hunk 2]
```diff
@@ -65,9 +65,9 @@ public String getDescription() {
...
         markEnd = i + 1;
-        if ("dhá".equalsIgnoreCase(tokens[i].getToken()) && (i < tokens.length - 2)) {
+        if ("dhá".equalsIgnoreCase(tokens[i].getToken())) {
```
Comment (TiagoSantos81) [inline languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java:70]: Supposedly redundant. This check here may be the most important one, but I still got IndexOutOfBounds without it.

File: languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java [hunk 3]
```diff
@@ -81,13 +81,13 @@ public String getDescription() {
...
           if (msg == null) {
-            msg = "Ba chóir duit <suggestion>" + tokens[i - 1].getToken() + " " + replacement + " " + tokens[i + 1].getToken() + "</suggestion> a scríobh";
+            msg = "Ba chóir duit <suggestion>" + replacement + "</suggestion> a scríobh";
           }
         }
       }
       if (msg != null) {
         RuleMatch match = new RuleMatch(
-          this, sentence, tokens[prevTokenIndex].getStartPos(), tokens[markEnd].getEndPos(), msg, "Uimhir phearsanta");
+          this, sentence, tokens[prevTokenIndex].getStartPos(), tokens[markEnd].getStartPos()+tokens[markEnd].getEndPos(), msg, "Uimhir phearsanta");
```
Comment (TiagoSantos81) [inline languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java:90]: `tokens[markEnd].getStartPos()+tokens[markEnd].getEndPos()` maybe I don't understand what you want here, but if thelast token is close to the sentence end position, adding something to `tokens[markEnd].getEndPos()` will also cause index out of bounds.
I am talking of this, after testing completed versions with the rule active. Without this, the rule can't be activated without crashing the program in the Irish language.

**Comment by TiagoSantos81** [comment]
Merged your changes with 167b430.

File: languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java (hunk 2 above)
Comment (jimregan) [reply languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java:70]: I changed the earlier test so it should have equivalent behaviour

File: languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java (hunk 3 above)
Comment (jimregan) [reply languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java:90]: getEndPos() gives the length of that token's match, not the overall, so it needs the start added from another token. I've put in a check to set it to 0 if it's in the same token

Comment (TiagoSantos81) [reply languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java:90]: You have looked at the internals better than I did, but, from experience, I believe that `getEndPos()` gives the numbered position in characters from the start of the sentence until the last character of the token. 
Anyway, please test the compiled version. I believe you will find it does what the tests expect (with a minor correction to the `beirt` suggestion).

Comment (jimregan) [reply languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java:90]: I was consistently getting assert errors that the getEndPos() result was less than getStartPos()

**Pull request merged**